File: javabackend/src/main/java/mft/dev/javabackend/dto/insert/response/TodoResponseDTO.java
package mft.dev.javabackend.dto.insert.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import mft.dev.javabackend.model.Todo;

public record TodoResponseDTO(
        @JsonProperty(value = "id", required = true)
        long id,
        @JsonProperty(value = "titolo", required = true)
        String titolo,
        @JsonProperty("descrizione")
        String descrizione
) {

    public static TodoResponseDTO fromTodo(Todo todo){
        return new TodoResponseDTO(
                todo.getId(),
                todo.getTitolo(),
                todo.getDescrizione()
        );
    }
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/dto/response/TodoResponseDTO.kt
package mft.dev.kotlinbackend.dto.response

import com.fasterxml.jackson.annotation.JsonProperty
import mft.dev.kotlinbackend.model.Todo

data class TodoResponseDTO(
    @JsonProperty(value = "id", required = true)
    val id: Long,
    @JsonProperty(value = "titolo", required = true)
    val titolo: String,
    @JsonProperty("descrizione")
    val descrizione: String?
) {
    companion object {
        fun fromTodo(todo: Todo): TodoResponseDTO {
            return TodoResponseDTO(
                todo.id!!,
                todo.titolo,
                todo.descrizione
            )
        }
    }
}
